scm-diff-editor/src/lib.rs

```diff
@@ -483,28 +487,32 @@ pub fn apply_changes(
         let file_path = write_root.join(file.path.clone());
         let (selected_contents, _unselected_contents) = file.get_selected_contents();
         match selected_contents {
-            SelectedContents::Absent => {
+            FileState::Absent => {
                 filesystem.remove_file(&file_path)?;
-            }
-            SelectedContents::Unchanged => {
-                // Do nothing.
-            }
-            SelectedContents::Binary {
-                old_description: _,
-                new_description: _,
-            } => {
-                let new_path = file_path;
-                let old_path = match &file.old_path {
-                    Some(old_path) => old_path.clone(),
-                    None => Cow::Borrowed(new_path.as_path()),
-                };
-                filesystem.copy_file(&old_path, &new_path)?;
-            }
-            SelectedContents::Present { contents } => {
-                if let Some(parent_dir) = file_path.parent() {
-                    filesystem.create_dir_all(parent_dir)?;
+            },
+            FileState::Present { contents, mode: _ } => {
```

Comment:
I might be missing something as it doesn't look like changes to the file's mode are handled as part of `apply_changes`